[FLINK-29020][docs] add document for CTAS feature (#20653)
apache/flink: docs/content/docs/dev/table/sql/create.md

**NOTE** The `source_table` can be a compound identifier. Thus, it can be a table from a different catalog or database: e.g. `my_catalog.my_db.MyTable` specifies table `MyTable` from catalog `MyCatalog` and database `my_db`; `my_db.MyTable` specifies table `MyTable` from current catalog and database `my_db`.

### `AS`

The AS clause is a variation of SQL features (Feature T172, “AS subquery clause in table definition”).

> The clause can be used to create a table based on the given query expression. It will be more user-friendly and reduce the cost for user to manually spell complex table creation statements.

Reviewer: Reference:
1. https://docs.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-develop-ctas#create-table-as-select
2. https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateTableAsSelect(CTAS)

Tables can also be created and populated by the results of a query in one create-table-as-select (CTAS) statement. CTAS is the simplest and fastest way to create and insert data into a table with a single command.

There are two parts in CTAS, the SELECT part can be any [SELECT statement](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/overview/) supported by Flink SQL. The CREATE part of the CTAS takes the resulting schema from the SELECT part and creates the target table with other table properties such as the connector and URL. Similar to CREATE TABLE, CTAS requires the required options of the corresponding connector must be specified in WITH clause.

> You can use this clause in stream and batch mode.

Reviewer: This is no need.

Consider the example statement below:

```sql
CREATE TABLE ctas_hudi
WITH (
    'connector' = 'hudi'
)
AS
SELECT id, name, age FROM test WHERE mod(id, 10) = 0;
```

The resulting table `ctas_hudi` will be equivalent to create the table and insert the data with the following statement:
```sql
CREATE TABLE ctas_hudi (
    id BIGINT,
    name STRING,
    age INT
) WITH (
     'connector' = 'hudi'
);
 
INSERT INTO ctas_hudi SELECT id, name, age FROM test WHERE mod(id, 10) = 0;
```

> **Notes:**

Reviewer: ```suggestion
**Note** CTAS has these restrictions:
```

> * does not support to create temporary table yet.

Reviewer: ```suggestion
* Does not support creating a temporary table yet.
```

> * does not support to specify explicit columns yet.

Reviewer: ```suggestion
* Does not support specifying explicit columns yet.
```

> * does not support to specify explicit watermark yet.

Reviewer: ```suggestion
* Does not support specifying explicit watermark yet.
```

> * does not support to create partitioned table yet.

Reviewer: ```suggestion
* Does not support creating partitioned table yet.
```

> * does not support primary key constraints yet.

Reviewer: ```suggestion
* Does not support specifying primary key constraints yet.
```

> **Notes:**

Reviewer: ```suggestion
**Note** The CTAS clause is currently non-atomicity and does not drop the target table when the job's final status is FAILED/CANCELED.
```

**NOTE** The `source_table` can be a compound identifier. Thus, it can be a table from a different catalog or database: e.g. `my_catalog.my_db.MyTable` specifies table `MyTable` from catalog `MyCatalog` and database `my_db`; `my_db.MyTable` specifies table `MyTable` from current catalog and database `my_db`.

### `AS`

Tables can also be created and populated by the results of a query in one create-table-as-select (CTAS) statement. 
CTAS is the simplest and fastest way to create and insert data into a table with a single command.

> There are two parts in CTAS, the SELECT part can be any [SELECT statement]({{< ref "docs/dev/table/sql/queries/overview" >}}) supported by Flink SQL. 

Reviewer: I think this would be more clear, WDYT?

```
Tables can also be created and populated by the results of a query in one create-table-as-select (CTAS) statement. CTAS is the simplest and fastest way to create and insert data into a table with a single command.

There are two parts in CTAS, the SELECT part can be any [SELECT query]({{< ref "docs/dev/table/sql/queries/overview" >}}) supported by Flink SQL. The CREATE part takes the resulting schema from the SELECT part and creates the target table. Similar to `CREATE TABLE`, CTAS requires the required options of the target table must be specified in WITH clause.

Creating the target table of CTAS depends on Catalog, so if using the built-in memory Catalog, users must ensure that the table already exists in external storage. If using other catalogs such as hive Catalog, the target table will be created by Catalog automatically.

```

The CREATE part of the CTAS takes the resulting schema from the SELECT part and creates the target table with other table properties such as the connector and URL. 
Similar to CREATE TABLE, CTAS requires the required options of the corresponding connector must be specified in WITH clause.

**NOTE** If using an in-memory catalog, the user must ensure that external storage already exists; If using an external catalog, then flink will create table from the external catalog, such as HiveCatalog.

Consider the example statement below:

```sql
CREATE TABLE my_ctas_table
WITH (
    'connector' = 'kafka',
    ...
)

> AS

Reviewer: ```suggestion
AS SELECT id, name, age FROM source_table WHERE mod(id, 10) = 0;
```

SELECT id, name, age FROM test WHERE mod(id, 10) = 0;
```

The resulting table `my_ctas_table` will be equivalent to create the table and insert the data with the following statement:
```sql
CREATE TABLE my_ctas_table (
    id BIGINT,
    name STRING,
    age INT
) WITH (
    'connector' = 'kafka',
    ...
);
 

> INSERT INTO my_ctas_table SELECT id, name, age FROM test WHERE mod(id, 10) = 0;

Reviewer: ```suggestion
INSERT INTO my_ctas_table SELECT id, name, age FROM source_table WHERE mod(id, 10) = 0;
```

```

**Note** CTAS has these restrictions:
* Does not support creating a temporary table yet.
* Does not support specifying explicit columns yet.
* Does not support specifying explicit watermark yet.
* Does not support creating partitioned table yet.
* Does not support specifying primary key constraints yet.

> **Note** The CTAS clause is currently non-atomicity and does not drop the target table when the job's final status is FAILED/CANCELED.

Reviewer: ```suggestion
**Note** The target table created by CTAS is non-atomic currently, the table won't be dropped automatically if occur errors while inserting data into the table.
```
